Lavellt account recovery

TuckerL-OSU opened the pull request.

tmrietz committed:
Add more cookies (user, admin)
  - Install ngx-cookie-service for parsing cookies on front-end
  - Use new parser for login redirect logic

tmrietz committed:
Cleanup cruft

TuckerL-OSU committed:
Added User Account Recovery via Recovery Questions

TODO:
Frontend not connected.
Logging doesn't work.

TuckerL-OSU committed:
Forgot to add a few files.
Added User Account Recovery via Recovery Questions

TODO:
Frontend not connected.
Logging doesn't work.

TuckerL-OSU committed:
Added User Account Recovery via Recovery Questions

TODO:
Logging doesn't work.
Need frontend password comparison/validation.

taylor-jones commented:
@TuckerL-OSU could you check on these merge conflicts before we look into merging?

TuckerL-OSU committed:
Rebased Account Recovery Stuff

Note: The Dang it error message pops up from the snackbar when you visit the /account page. https://i.imgur.com/AcTZZLt.png
